### DISCUSSION TITLE
feature: Add support for saving object as well when option from select dropdown is selected.

### DISCUSSION THREAD

**Discussion by ahmedsakri** [manojadams/mui-forms#44] [Ideas]
**Is your feature request related to a problem? Please describe.**
I am doing an API call in select dropdown to load options. The api returns below response:
```
[
    {
        "id": 1,
        "name": "John Doe",
        "email": "johndoe@gmail.com"
    },
    {
        "id": 2,
        "name": "Jane Doe",
        "email": "janedoe@gmail.com"
    }
]
```

Json Schema for select looks like below:
```
{
  "name": "sample",
  "meta": {
    "config": {
      "requestType": "get",
      "urlType": "remote",
      "url": "http://localhost:3008/v1.0/default/custom",
      "labelKey": "name",
      "valueKey": "id"
    }
  }
}

```

By having above schema, what happens is only `id` is getting saved alongside the keyname. 
```
sample: 1 or sample: 2
```

**Describe the solution you'd like**
A way i can ask to store an object in the `sample` key, user can store the whole object or selected keys!

```
"sample": {
        "id": 1,
        "name": "John Doe",
        "email": "johndoe@gmail.com"
    }
```

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by manojadams** [comment]
@ahmedsakri Do you mean you want the whole response object to be present on click of submit button.

**Comment by ahmedsakri** [comment]
@manojadams 
It can either be the whole object, or it user wants to save both, `id and name` as a value for sample key.

So, `sample` key is the select dropdown. in my mentioned case. the value of it will be:

```
sample: {
  name: "John Doe",
  id: 1
}
```

**Comment by manojadams** [comment]
@ahmedsakri I am not entirely convinced about this.
I will appreciate if you can suggest any other alternative way or existing example to show this.

Request to add any reference to check similar examples/requirements.

**Comment by manojadams** [comment]
@ahmedsakri This is now supported and can be done through value dependency.
Ref:
- https://mui-forms.vercel.app/field-dependencies/value
- https://mui-forms.vercel.app/examples/registration-form
